### PR TITLE
Add $ token

### PR DESCRIPTION
**Pull request by MSleepyPanda**
Closes https://github.com/dtolnay/syn/issues/419

### CODE REVIEW

**Comment by MSleepyPanda** [comment]
Interesting, this errors on Rust 1.15.1 but is fine on every other version, including recent stable, beta and nightly

**Comment by MSleepyPanda** [comment]
Sure thing! I've uncommented it and left a ToDo

File: src/token.rs [hunk 1]
```diff
@@ -584,6 +586,7 @@ macro_rules! punct {
     ($i:expr, ,)   => { call!($i, <$crate::token::Comma as $crate::synom::Synom>::parse) };
     ($i:expr, /)   => { call!($i, <$crate::token::Div as $crate::synom::Synom>::parse) };
     ($i:expr, /=)  => { call!($i, <$crate::token::DivEq as $crate::synom::Synom>::parse) };
+    ($i:expr, $)   => { call!($im <$crate::token::Dollar as $crate::synom::Synom>::parse) };
```
Comment (dtolnay) [inline src/token.rs]: This line will not compile on 1.15.0 and 1.16.0.

File: src/token.rs [hunk 2]
```diff
@@ -484,6 +485,7 @@ macro_rules! Token {
     (,)        => { $crate::token::Comma };
     (/)        => { $crate::token::Div };
     (/=)       => { $crate::token::DivEq };
+//  ($)        => { $crate::token::Dollar }; //TODO: Enable when upgrading to Rust >= 1.16
```
Comment (dtolnay) [inline src/token.rs]: I would prefer to remove these commented lines and just file a GitHub issue to follow up when we eventually bump the required compiler version. Having an issue makes it easier to label and prioritize and track the work.

File: src/token.rs [hunk 3]
```diff
@@ -376,6 +376,7 @@ tokens! {
         ","        pub struct Comma/1      /// `,`
         "/"        pub struct Div/1        /// `/`
         "/="       pub struct DivEq/2      /// `/=`
+//      "$"        pub struct Dollar/1     /// `$` //TODO: Enable when upgrading to Rust >= 1.16
```
Comment (dtolnay) [inline src/token.rs]: Please uncomment this line.

**Comment by MSleepyPanda** [comment]
Updated! Sorry, i realized that i've uncommented the wrong line the second i pressed comment 😅

**Comment by dtolnay** [comment]
I released syn 0.13.7 with this change.

**Pull request merged**